### PR TITLE
Dependency graph: use virtual paths

### PR DESCRIPTION
**Pull request by rhelmke**
_This PR is supplementary to PR #726 and a follow-up of PR #725 ._

It fixes the dependency graph to properly display symlinks. Multiple links on a single file (i.e., in busybox-centric environments) produce the same object hash. Thus, FACT aggregates them in a single file object for deduplication and adds a virtual file path reference in the dataset.

As the previously reworked dependency graph created nodes based on file objects instead of virtual file paths, it did not incorporate all nodes existing in the path structure.

This PR fixes this issue.

### CODE REVIEW

File: src/web_interface/components/dependency_graph.py [hunk 1]
```diff
@@ -13,17 +15,36 @@ def create_data_graph_nodes_and_groups(data, whitelist):
...
+            if parent_component != parent_uid:
+                continue
+
             node = {
-                'label': file['file_name'],
-                'id': file['_id'],
+                'label': name_component,
+                'id': vpath,
+                'entity': file['_id'],
                 'group': file['processed_analysis']['file_type']['mime'],
```
Comment (jstucke) [inline src/web_interface/components/dependency_graph.py:44]: can also be replaced:
```suggestion
                'group': mime,
```

File: src/web_interface/templates/show_analysis.html [hunk 2]
```diff
@@ -64,7 +64,7 @@
                 {% if not firmware.files_included %}
                     {{ button_tooltip('Show dependency graph', 'graph-button', '/dependency-graph/', 'project-diagram', danger=False, disabled=True) }}
                 {% else %}
-                    {{ button_tooltip('Show dependency graph', 'graph-button', '/dependency-graph/', 'project-diagram') }}
+                    {{ button_tooltip('Show dependency graph', 'graph-button', '/dependency_graph/', 'project-diagram', 'window.location.href = \'/dependency-graph/' + '/' + firmware.uid + '/' + root_uid + '\'') }}
```
[the remark this answers is not in the document]
Comment (rhelmke) [reply src/web_interface/templates/show_analysis.html:67]: solved

**Review by jstucke — COMMENTED** [review]
Seems to work fine.

File: src/web_interface/components/dependency_graph.py [hunk 3]
```diff
@@ -52,21 +60,20 @@ def create_data_graph_nodes_and_groups(data, parent_uid, root_uid, whitelist):
...
+        if node['elf_analysis_missing']:
+            elf_analysis_missing_from_files += 1
             continue
 
-        for lib in libraries:
-            edge_id = find_edges(data_graph, edge_id, lib, file)
+        linked_libraries = node['linked_libraries']
+
+        for linked_lib_name in linked_libraries:
+            edge_id = find_edges(node, linked_lib_name, data_graph, edge_id)
```
[the remark this answers is not in the document]
Comment (rhelmke) [reply src/web_interface/components/dependency_graph.py:76]: Yes, yes indeed. I was really frustrated about that as well.

**Review by jstucke — APPROVED** [review]

**Pull request merged**